core/drivers/firewall/stm32_rifsc.c

```diff
@@ -0,0 +1,563 @@
+#define RIFSC_RISC_PER_ID_MASK		GENMASK_32(31, 24)
+
+#define RIFSC_RISC_PERx_CID_MASK	(RIFSC_RISC_CFEN_MASK | \
+					 RIFSC_RISC_SEM_EN_MASK | \
+					 RIFSC_RISC_SCID_MASK | \
+					 RIFSC_RISC_SEML_MASK)
+
+#define RIFSC_RIMC_MODE_MASK		BIT(2)
+#define RIFSC_RIMC_MCID_MASK		GENMASK_32(6, 4)
+#define RIFSC_RIMC_MSEC_MASK		BIT(8)
+#define RIFSC_RIMC_MPRIV_MASK		BIT(9)
+#define RIFSC_RIMC_M_ID_MASK		GENMASK_32(23, 16)
+
+#define RIFSC_RIMC_ATTRx_MASK		(RIFSC_RIMC_MODE_MASK | \
+					 RIFSC_RIMC_MCID_MASK | \
+					 RIFSC_RIMC_MSEC_MASK | \
+					 RIFSC_RIMC_MPRIV_MASK)
+
+/* max entries */
+#define MAX_RIMU			U(16)
+#define MAX_RISUP			U(128)
+
+#define _RIF_FLD_PREP(field, value)	(((uint32_t)(value) << \
+					  (field ## _SHIFT)) & (field ## _MASK))
+#define _RIF_FLD_GET(field, value)	(((uint32_t)(value) & \
+					  (field ## _MASK)) >>\
+					 (field ## _SHIFT))
+
+struct rifsc_driver_data {
+	uint32_t version;
+	bool rif_en;
+	bool sec_en;
+	bool priv_en;
+	uint8_t nb_rimu;
+	uint8_t nb_risup;
+	uint8_t nb_risal;
+};
+
+struct rifsc_platdata {
+	uintptr_t base;
+	struct rifsc_driver_data *drv_data;
+	struct risup_cfg *risup;
+	int nrisup;
+	struct rimu_cfg *rimu;
+	int nrimu;
+};
+
+struct dt_id_attr {
+	/* The effective size of the array is meaningless here */
+	fdt32_t id_attr[1];
+};
+
+static struct rifsc_driver_data rifsc_drvdata;
+static struct rifsc_platdata rifsc_pdata;
+
+static void stm32_rifsc_get_driverdata(struct rifsc_platdata *pdata)
+{
+	uint32_t regval = 0;
+
+	regval = io_read32(pdata->base + _RIFSC_HWCFGR1);
+	rifsc_drvdata.rif_en = _RIF_FLD_GET(_RIFSC_HWCFGR1_CFG1, regval) != 0;
+	rifsc_drvdata.sec_en = _RIF_FLD_GET(_RIFSC_HWCFGR1_CFG2, regval) != 0;
+	rifsc_drvdata.priv_en = _RIF_FLD_GET(_RIFSC_HWCFGR1_CFG3, regval) != 0;
+
+	regval = io_read32(pdata->base + _RIFSC_HWCFGR2);
+	rifsc_drvdata.nb_risup = _RIF_FLD_GET(_RIFSC_HWCFGR2_CFG1, regval);
+	rifsc_drvdata.nb_rimu = _RIF_FLD_GET(_RIFSC_HWCFGR2_CFG2, regval);
+	rifsc_drvdata.nb_risal = _RIF_FLD_GET(_RIFSC_HWCFGR2_CFG3, regval);
+
+	pdata->drv_data = &rifsc_drvdata;
+
+	regval = io_read8(pdata->base + _RIFSC_VERR);
+
+	DMSG("RIFSC version %"PRIu32".%"PRIu32,
+	     _RIF_FLD_GET(_RIFSC_VERR_MAJREV, regval),
+	     _RIF_FLD_GET(_RIFSC_VERR_MINREV, regval));
+
+	DMSG("HW cap: enabled[rif:sec:priv]:[%s:%s:%s] nb[risup|rimu|risal]:[%"PRIu8",%"PRIu8",%"PRIu8"]",
+	     rifsc_drvdata.rif_en ? "true" : "false",
+	     rifsc_drvdata.sec_en ? "true" : "false",
+	     rifsc_drvdata.priv_en ? "true" : "false",
+	     rifsc_drvdata.nb_risup,
+	     rifsc_drvdata.nb_rimu,
+	     rifsc_drvdata.nb_risal);
+}
+
+static TEE_Result stm32_rifsc_glock_config(const void *fdt, int node,
+					   struct rifsc_platdata *pdata)
+{
+	const fdt32_t *cuint = NULL;
+	int len = 0;
+	uint32_t glock_conf = 0;
+
+	cuint = fdt_getprop(fdt, node, "st,glocked", &len);
+	if (!cuint) {
+		DMSG("No global lock on RIF configuration");
+		return TEE_SUCCESS;
+	}
+
```

Comment:
You could remove this empty line as this assert is straight related to the instruction right above.